ci/pipelines/apps-test-ecr.yml

```diff
@@ -16,6 +16,17 @@ aws_staging_config: &aws_staging_config
   aws_ecr_registry_id: "((pay_aws_staging_account_id))"
   aws_region: eu-west-1
 
+wait_params: &wait_params
+  AWS_ACCESS_KEY_ID: ((.:role.AWS_ACCESS_KEY_ID))
+  AWS_SECRET_ACCESS_KEY: ((.:role.AWS_SECRET_ACCESS_KEY))
+  AWS_SESSION_TOKEN: ((.:role.AWS_SESSION_TOKEN))
+  TAG: ((.:tag))
+  ENVIRONMENT: test-12
```

Comment:
that wouldn't work though - it would overwrite the params object

Reply:
or could rename to `wait_for_deploy_params` if that helps?

Reply:
Oh it overwrites the params object? ewww. am happy with your `aws_assumed_role_creds: &aws_assumed_role_creds` suggestion, if you make that change I'll approve it

But looking at the pull-image-from-dockerhub example again I think this would work, but I'll leave it to your discretion.
```
definitions:
  - &wait_for_deploy
    task: wait-for-deploy
    file: pay-ci/ci/tasks/wait-for-deploy.yml
    params:
      AWS_ACCESS_KEY_ID: updateThisValue
      AWS_SECRET_ACCESS_KEY: updateThisValue
      AWS_SESSION_TOKEN: updateThisValue
      APP_NAME: updateThisValue
      TAG: updateThisValue

aws_assumed_role_creds: &aws_assumed_role_creds
  AWS_ACCESS_KEY_ID: ((.:role.AWS_ACCESS_KEY_ID))
  AWS_SECRET_ACCESS_KEY: ((.:role.AWS_SECRET_ACCESS_KEY))
  AWS_SESSION_TOKEN: ((.:role.AWS_SESSION_TOKEN))
```
then
```
- name: deploy-toolbox
  plan:
    .
    .
    .
    etc
    .
    - <<: *wait_for_deploy
      params:
        APP_NAME: selfservice
        TAG: ((.:tag))
        <<: * aws_assumed_role_creds
```

Reply:
`wait_for_deploy_params` works too.

Reply:
yeah we can definitely have a template for the task, but there's kinda no point having the params in the task template as you just have to overwrite them every time (I guess the point is to be clear what is needed)

Reply:
I actually chose not to template out the whole task as I find it easier to read without all the anchors at the job level (see the pr pipeline for where we probably went overboard with this). So I think I'll go with `wait_for_deploy_params` for now if that's ok

Reply:
Yeah I was thinking more about extracting the `file: pay-ci/ci/tasks/wait-for-deploy.yml` into one place! Happy with whatever you choose though.